tests/test_doctestplus.py

```diff
@@ -392,3 +392,59 @@ def test_requires(testdir):
     )
     # passed because 'pytest<1.0' was not satisfied and 'assert 0' was not evaluated
     testdir.inline_run(p, '--doctest-plus', '--doctest-rst').assertoutcome(passed=1)
+
+
+def test_ignore_warnings_module(testdir):
+
+    # First check that we get a warning if we don't add the IGNORE_WARNINGS
+    # directive
+    p = testdir.makepyfile(
+        """
+        def myfunc():
+            '''
+            >>> import warnings
+            >>> warnings.warn('A warning occurred', UserWarning)
+            '''
+            pass
+        """)
+    reprec = testdir.inline_run(p, "--doctest-plus", "-W error")
+    reprec.assertoutcome(failed=1, passed=0)
+
+    # Now try with the IGNORE_WARNINGS directive
+    p = testdir.makepyfile(
+        """
+        def myfunc():
+            '''
+            >>> import warnings
+            >>> warnings.warn('A warning occurred', UserWarning)  # doctest: +IGNORE_WARNINGS
+            '''
+            pass
+        """)
+    reprec = testdir.inline_run(p, "--doctest-plus", "-W error")
+    reprec.assertoutcome(failed=0, passed=1)
+
+
+def test_ignore_warnings_rst(testdir):
+
+    # First check that we get a warning if we don't add the IGNORE_WARNINGS
+    # directive
+    p = testdir.makefile(".rst",
+        """
+        ::
+            >>> import warnings
+            >>> warnings.warn('A warning occurred', UserWarning)
+        """)
+    reprec = testdir.inline_run(p, "--doctest-plus", "--doctest-rst",
+                                "--text-file-format=rst", "-W error")
+    reprec.assertoutcome(failed=1, passed=0)
+
+    # Now try with the IGNORE_WARNINGS directive
+    p = testdir.makefile(".rst",
+        """
+        ::
+            >>> import warnings
+            >>> warnings.warn('A warning occurred', UserWarning)  # doctest: +IGNORE_WARNINGS
+        """)
+    reprec = testdir.inline_run(p, "--doctest-plus", "--doctest-rst",
+                                "--text-file-format=rst", "-W error")
```

Comment:
Flag should work without `-W error`, right?

Reply:
This was to make sure that the warning triggers a failure in the tests - for older pytest versions this isn't the case.